internal/rpcproxy/keymanager.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Canonical.
+
+package rpcproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"regexp"
+
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/utils/v3/ssh"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/canonical/jimm/v3/internal/jimm/sshkeys"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+var isFingerprintRegexp = regexp.MustCompile("^[0-9a-f]{2}(:[0-9a-f]{2}){15}$")
+
+// keyManagerFacade is intended to be a temporary struct used to emulate logic
+// that will eventually live in jujuapi. This struct contains all
+// the api layer logic for SSH key management methods that are currently
+// used by the rpcProxy.
+type keyManagerFacade struct {
+	SSHKeyManager
+	user *openfga.User
+}
+
+// ListKeys lists the authenticated user's SSH keys
+// in the format defined in args.
+func (s *keyManagerFacade) ListKeys(ctx context.Context, args jujuparams.ListSSHKeys) (jujuparams.StringsResults, error) {
+	keys, err := s.ListUserPublicKeys(ctx, s.user)
+	if err != nil {
+		return jujuparams.StringsResults{}, err
+	}
+
+	var formatter func(key sshkeys.PublicKey) string
+	switch args.Mode {
+	case ssh.FullKeys:
+		formatter = marshalAuthorizedKeyWithComment
+	case ssh.Fingerprints:
+		formatter = fingerprintWithComment
+	default:
+		return jujuparams.StringsResults{}, fmt.Errorf("unknown mode (%v)", args.Mode)
+	}
+
+	// The Juju CLI takes the first element from the slice of stringsResults.
+	res := jujuparams.StringsResult{}
+	for _, key := range keys {
+		res.Result = append(res.Result, formatter(key))
+	}
+	return jujuparams.StringsResults{Results: []jujuparams.StringsResult{res}}, nil
+}
+
+// AddKeys saves the SSH keys defined in args and associates them
+// with the authenticated user.
+func (s *keyManagerFacade) AddKeys(ctx context.Context, args jujuparams.ModifyUserSSHKeys) (jujuparams.ErrorResults, error) {
+	var res []jujuparams.ErrorResult
+	errF := func(err error, msg string) jujuparams.ErrorResult {
```

Comment:
suggestion: just have `func(msg string) jujuparams.ErrorResult`.. you are already creating a message string every time you call `errF`, so appending the `err.Error()` each time, would be a minor change.

Reply:
then perhaps 
```
func(msg, errorCode string) jujuparams.ErrorResult {
  		return jujuparams.ErrorResult{Error: &jujuparams.Error{
			Code:    string(errors.ErrorCode(err)),
			Message: msg,
		}}
}
```
but just a suggestion, i won't insist on it..

Reply:
It's minor so I've left it as is.